### ISSUE TITLE
Doubts regarding A* heuristic

### ISSUE THREAD

**Issue by DieterDePaepe** [RubenVerborgh/Computing-Interesting-Paths#2]
On slide 22, it is mentioned that the Jaccard distance is used as a heuristic.

Given 2 nodes `V` and `U`, where `V` has 1000 different predicates connected to it, and `U` has 1 unique predicate connected to it. The heuristic would estimate:

```
goalDistance = 1 - |1| / |1000| = 0.999
```

However, the actual weight between `V` and `U` is based on the occurrent of the connecting predicate in the remainder of the graph. As such, the heuristic might overestimate the actual distance. Although this is technically allowed in A*, it might result in paths that are longer than the actual shortest path.

**Comment by RubenVerborgh** [comment]
Overestimation is fine in this case; having a more interesting path is more important than having the shortest path. Special here is that the weight function and the heuristic have conflicting interests: the weight function wants the path to be as interesting (= difficult) as possible, whereas the heuristic wants to get things done fast.

**Comment by DieterDePaepe** [comment]
Overestimation can result in a path that is not interesting at all. Since a short path is an interesting path, an overly long path can mean a non interesting path.

**Comment by RubenVerborgh** [comment]
> Overestimation can result in a path that is not interesting at all.

True; but note that even a "perfect" estimation does not necessarily lead to an interesting path.

> Since a short path is an interesting path

Not necessarily! In fact, this was one of our early problems. For instance, London and Paris would be connected with "London is a city, city is the type of Paris", which is possibly the least interesting path possible 😄 

> an overly long path can mean a non interesting path

Also true; but overestimation does not necessarily lead to _long_ paths; it just removes the guarantee that the path will be the shortest.
